workers/data_refinery_workers/downloaders/array_express.py

```diff
@@ -33,20 +30,16 @@ def download_array_express(job_id):
         utils.end_job(job, None, False)
         return
 
-    target_directory = ROOT_URI + batch.internal_location
-    os.makedirs(target_directory, exist_ok=True)
-
-    filename = batch.download_url.split('/')[-1]
-    target_file_name = target_directory + filename
+    target_file_path = utils.prepare_destination(batch)
 
     logger.info("Downloading file from %s to %s. (Batch #%d, Job #%d)",
                 batch.download_url,
-                target_file_name,
+                target_file_path,
                 batch.id,
                 job_id)
 
     try:
-        target_file = open(target_file_name, "wb")
+        target_file = open(target_file_path, "wb")
         request = requests.get(batch.download_url, stream=True)
```

Comment:
It's good that you have a `target_file.close()` in the `finally` block below, but check to see if you also need to clean up the connection created by `requests.get(..., stream=True)`. See:

> http://docs.python-requests.org/en/master/user/advanced/#body-content-workflow

where it says "If you set stream to True when making a request, Requests cannot release the connection back to the pool unless you consume all the data or call Response.close."